.github/workflows/tdd-pipeline.yml

```
@@ -0,0 +1,305 @@
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: report
+            });
+
+  # 🚀 Deploy Preview (on successful PR)
+  deploy-preview:
+    name: 🚀 Deploy Preview
+    runs-on: ubuntu-latest
+    needs: quality-metrics
+    if: github.event_name == 'pull_request' && needs.quality-gate.result == 'success'
```

actionlint (1.7.7)

property "quality-gate" is not defined in object type {quality-metrics: {outputs: {}; result: string}}

(expression)